src/Internal/QueryM/Sign.purs

```diff
@@ -5,32 +5,23 @@ module Ctl.Internal.QueryM.Sign
 import Prelude
 
 import Control.Monad.Reader (asks)
-import Ctl.Internal.Cardano.Types.Transaction (_body, _inputs, _witnessSet)
+import Ctl.Internal.Cardano.Types.Transaction (_witnessSet)
 import Ctl.Internal.Cardano.Types.Transaction as Transaction
-import Ctl.Internal.Helpers (liftedM)
 import Ctl.Internal.QueryM
   ( QueryM
   , callCip30Wallet
-  , getWalletAddresses
   , withMWallet
   )
-import Ctl.Internal.QueryM.Utxos (getUtxo, getWalletUtxos)
-import Ctl.Internal.Types.Transaction (TransactionInput)
 import Ctl.Internal.Wallet
   ( Wallet(KeyWallet, Lode, Eternl, Flint, Gero, Nami, NuFi)
   )
-import Data.Array (elem, fromFoldable)
 import Data.Lens ((<>~))
-import Data.Lens.Getter ((^.))
-import Data.Map as Map
-import Data.Maybe (Maybe(Just), fromMaybe)
-import Data.Newtype (unwrap, wrap)
-import Data.Traversable (for_, traverse)
-import Data.Tuple.Nested ((/\))
-import Effect.Aff (delay, error)
+import Data.Maybe (Maybe(Just))
+import Data.Newtype (unwrap)
+import Data.Traversable (for_)
 import Effect.Aff.Class (liftAff)
 import Effect.Class (liftEffect)
-import Effect.Exception (throw, try)
+import Effect.Exception (try)
 
 signTransaction
```

Comment:
I'm tempted to just remove this module and put `signTransaction` under `QueryM` and use `actionBasedOnWallet` to make it shorter.